src/main/java/ladder/model/Line.java

```diff
@@ -21,38 +20,34 @@ public Line(int countOfStep, LadderGenerateStrategy generateStrategy) {
         }
     }
 
-    private boolean hasBeforeStep(int pointIndex) {
-        return pointIndex > 0 && steps.get(pointIndex - STEP_INTERVAL);
-    }
-
-    private void validateLine(List<Boolean> steps) {
-        boolean hasInvalidStep = IntStream.range(MIN_INDEX, steps.size())
-                                        .anyMatch(i -> steps.get(i) && hasBeforeStep(i));
-
-        if (hasInvalidStep) {
-            throw new IllegalArgumentException("인접한 사다리 다리가 있습니다.");
-        }
-    }
-
-    public List<Boolean> getSteps() {
-        return Collections.unmodifiableList(steps);
+    private boolean hasBeforeStep(int position) {
+        validateIndex(position);
+        return position > 0 && steps.get(position - STEP_INTERVAL);
     }
 
     private boolean hasNextStep(int pointIndex) {
         validateIndex(pointIndex);
-        return steps.get(pointIndex);
+        return pointIndex < steps.size() && steps.get(pointIndex);
     }
 
     private void validateIndex(int pointIndex) {
-        if (pointIndex < 0 || pointIndex >= steps.size()) {
+        if (pointIndex < 0 || pointIndex > steps.size()) {
             throw new IllegalArgumentException("index가 사다리 범위를 벗어났습니다.");
         }
     }
 
-    public void processLine(Users users) {
-        IntStream.range(0, steps.size())
-                .filter(this::hasNextStep)
-                .forEach(i -> users.swapUserPoint(i, i + 1));
+    public List<Boolean> getSteps() {
+        return Collections.unmodifiableList(steps);
+    }
+
+    public int movePoint(int pointIndex) {
+        if (hasNextStep(pointIndex)) {
+            return pointIndex + 1;
```

Comment:
```suggestion
            return moveRight();
```
와 같이 의미를 부여할수 있을것 같습니다 :)

Reply:
네 ㅎㅎ 훨씬 명확하네요👍